File: src/main/java/cn/mxleader/quickdoc/security/session/ActiveUserStore.java
package cn.mxleader.quickdoc.security.session;

import org.springframework.beans.factory.annotation.Autowired;

import java.util.ArrayList;
import java.util.List;

public class ActiveUserStore {

    public List<String> users;

    @Autowired
    public ActiveUserStore() {
        users = new ArrayList<String>();
    }

    public List<String> getUsers() {
        return users;
    }

    public void setUsers(List<String> users) {
        this.users = users;
    }

    public void addUser(String username){
        users.add(username);
    }

    public void removeUser(String username){
        users.remove(username);
    }
}

File: src/main/kotlin/cn/mxleader/quickdoc/web/session/ActiveUserStore.kt
package cn.mxleader.quickdoc.web.session

data class ActiveUserStore(var users:List<String>) {
    constructor():this(emptyList<String>())

    fun addUser(username: String) {
        users += username
    }

    fun removeUser(username: String) {
        users -= username
    }
}